+++ readline-6.3/aclocal.m4	2013-10-21 07:37:50.000000000 +0900
--- readline-6.3-fixed/aclocal.m4	2015-03-26 00:21:13.517479014 +0900
@@ -1784,7 +1784,7 @@
         exit (w == 0);  /* exit 0 if wcwidth broken */
 }
 ],
-bash_cv_wcwidth_broken=yes, bash_cv_wcwdith_broken=no, bash_cv_wcwidth_broken=no)])
+bash_cv_wcwidth_broken=yes, bash_cv_wcwidth_broken=no, bash_cv_wcwidth_broken=no)])
 if test "$bash_cv_wcwidth_broken" = yes; then
         AC_DEFINE(WCWIDTH_BROKEN, 1, [wcwidth is usually not broken])
 fi
--- readline-6.3/configure	2013-03-13 23:14:53.000000000 +0900
+++ readline-6.3-fixed/configure	2015-03-26 00:25:32.378912080 +0900
@@ -6352,7 +6352,7 @@
 if ac_fn_c_try_run "$LINENO"; then :
   bash_cv_wcwidth_broken=yes
 else
-  bash_cv_wcwdith_broken=no
+  bash_cv_wcwidth_broken=no
 fi
 rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
   conftest.$ac_objext conftest.beam conftest.$ac_ext
