diff -ru Python-3.3.5/Makefile.pre.in Python-3.3.5-android/Makefile.pre.in
--- Python-3.3.5/Makefile.pre.in	2014-03-09 09:40:23.000000000 +0100
+++ Python-3.3.5-android/Makefile.pre.in	2014-08-04 22:13:00.000000000 +0200
@@ -1243,6 +1243,7 @@
 # Install the dynamically loadable modules
 # This goes into $(exec_prefix)
 sharedinstall: sharedmods
+	CC='$(CC)' LDSHARED='$(BLDSHARED)' LDFLAGS='$(LDFLAGS)' OPT='$(OPT)' CROSS_COMPILE='$(CROSS_COMPILE)' \
 	$(RUNSHARED) $(PYTHON_FOR_BUILD) $(srcdir)/setup.py install \
 	   	--prefix=$(prefix) \
 		--install-scripts=$(BINDIR) \
