diff -ur python-src_arm/Python/bltinmodule.c.orig python-src_arm/Python/bltinmodule.c
--- python-src_arm/Python/bltinmodule.c.orig	2017-01-10 03:07:25.303707373 +0900
+++ python-src_arm/Python/bltinmodule.c	2017-01-10 03:17:10.587231203 +0900
@@ -21,7 +21,7 @@
    Don't forget to modify PyUnicode_DecodeFSDefault() if you touch any of the
    values for Py_FileSystemDefaultEncoding!
 */
-#if defined(__APPLE__)
+#if defined(__APPLE__) || defined(__ANDROID__)
 const char *Py_FileSystemDefaultEncoding = "utf-8";
 int Py_HasFileSystemDefaultEncoding = 1;
 #elif defined(MS_WINDOWS)
