diff -ru Python-3.3.5/Lib/test/test_pwd.py Python-3.3.5-android/Lib/test/test_pwd.py
--- Python-3.3.5/Lib/test/test_pwd.py	2014-03-09 09:40:19.000000000 +0100
+++ Python-3.3.5-android/Lib/test/test_pwd.py	2014-08-04 22:14:36.000000000 +0200
@@ -6,6 +6,7 @@

 class PwdTest(unittest.TestCase):

+ @unittest.skipUnless(hasattr(pwd, 'getpwall'), 'pwd module does not expose getpwall()')
     def test_values(self):
         entries = pwd.getpwall()

@@ -52,6 +53,7 @@
             self.assertIn(pwd.getpwnam(e.pw_name), entriesbyname[e.pw_name])
             self.assertIn(pwd.getpwuid(e.pw_uid), entriesbyuid[e.pw_uid])

+ @unittest.skipUnless(hasattr(pwd, 'getpwall'), 'pwd module does not expose getpwall()')
     def test_errors(self):
         self.assertRaises(TypeError, pwd.getpwuid)
         self.assertRaises(TypeError, pwd.getpwuid, 3.14)
diff -ru Python-3.3.5/setup.py Python-3.3.5-android/setup.py
--- Python-3.3.5/setup.py	2014-03-09 09:40:35.000000000 +0100
+++ Python-3.3.5-android/setup.py	2014-08-04 22:14:36.000000000 +0200
@@ -574,7 +577,7 @@
                                libraries=math_libs) )

         # time libraries: librt may be needed for clock_gettime()
-        time_libs = []
+        time_libs = ['m']
         lib = sysconfig.get_config_var('TIMEMODULE_LIB')
         if lib:
             time_libs.append(lib)
@@ -631,7 +631,9 @@
             missing.append('spwd')
 
         # select(2); not on ancient System V
-        exts.append( Extension('select', ['selectmodule.c']) )
+        # selecemodule.c calls the ceil(3) math function.
+        exts.append(Extension('select', ['selectmodule.c'],
+                              libraries=['m']))
 
         # Fred Drake's interface to the Python parser
         exts.append( Extension('parser', ['parsermodule.c']) )
@@ -1339,7 +1336,10 @@
         #
         # You can upgrade zlib to version 1.1.4 yourself by going to
         # http://www.gzip.org/zlib/
-        zlib_inc = find_file('zlib.h', [], inc_dirs)
+        sysroot = os.environ["SYSROOT"]
+        _inc_dirs = inc_dirs + [sysroot + "/usr/include"]
+        _lib_dirs = lib_dirs + [sysroot + "/usr/lib"]
+        zlib_inc = find_file('zlib.h', [], _inc_dirs)
         have_zlib = False
         if zlib_inc is not None:
             zlib_h = zlib_inc[0] + '/zlib.h'
@@ -1356,7 +1356,7 @@
                         version = line.split()[2]
                         break
             if version >= version_req:
-                if (self.compiler.find_library_file(lib_dirs, 'z')):
+                if (self.compiler.find_library_file(_lib_dirs, 'z')):
                     if host_platform == "darwin":
                         zlib_extra_link_args = ('-Wl,-search_paths_first',)
                     else:
