diff -uprN pycrypto-2.3/setup.cfg pycrypto-2.3.mine//setup.cfg
--- pycrypto-2.3/setup.cfg	1969-12-31 21:00:00.000000000 -0300
+++ pycrypto-2.3.mine//setup.cfg	2011-02-14 17:35:25.000000000 -0300
@@ -0,0 +1,5 @@
+[bdist_egg]
+plat-name=linux-armv
+
+[build_ext]
+include-dirs=bluez
diff -uprN pycrypto-2.3/setup.py pycrypto-2.3.mine//setup.py
--- pycrypto-2.3/setup.py	2010-08-27 01:41:55.000000000 -0300
+++ pycrypto-2.3.mine//setup.py	2011-05-26 14:00:00.000000000 -0300
@@ -37,6 +37,10 @@
 __revision__ = "$Id$"
 
 from distutils import core
+from py4a import patch_distutils
+patch_distutils()
+
+from setuptools import setup
 from distutils.core import Extension, Command
 from distutils.command.build_ext import build_ext
 import os, sys
@@ -337,5 +341,5 @@ if hasattr(core, 'setup_keywords'):
         kw['download_url'] = ('http://www.pycrypto.org/files/'
                               '%s-%s.tar.gz' % (kw['name'], kw['version']) )
 
-core.setup(**kw)
+setup(**kw)
 
