diff --ignore-file-name-case -wuprN Python-2.7.8/Modules/_localemodule.c python-src/Modules/_localemodule.c
--- Python-2.7.8/Modules/_localemodule.c	2014-06-30 11:05:41.000000000 +0900
+++ python-src/Modules/_localemodule.c	2015-03-15 08:59:51.830684221 +0900
@@ -205,6 +205,7 @@ PyDoc_STRVAR(localeconv__doc__,
 static PyObject*
 PyLocale_localeconv(PyObject* self)
 {
+#ifndef ANDROID
     PyObject* result;
     struct lconv *l;
     PyObject *x;
@@ -265,6 +266,7 @@ PyLocale_localeconv(PyObject* self)
   failed:
     Py_XDECREF(result);
     Py_XDECREF(x);
+#endif
     return NULL;
 }
 

